### PR TITLE
Creates a script to get the age for each direct dependency

### PR DESCRIPTION
**Pull request by kyallanum-MND**

### CODE REVIEW

File: Scripts/Mend SCA/get-library-age.sh [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+		echo "Logging in with provided API key."
+		LOGIN_BODY="{\"email\": \"$MEND_EMAIL\", \"userKey\": \"$MEND_USER_KEY\", \"orgToken\": \"$WS_APIKEY\"}"
+	fi
+}
+
+function api_login() {
+	# Log into API 2.0 and get the JWT Token and Organization UUID
+	echo $LOGIN_BODY
+	LOGIN_RESPONSE=$(curl -s -X POST --location "$MEND_API_URL/login" --header 'Content-Type: application/json' --data-raw "${LOGIN_BODY}")
+	echo $LOGIN_RESPONSE
```
Comment (samq-ws) [inline Scripts/Mend SCA/get-library-age.sh:63]: These echos seems like leftover debug code

**Review by samq-ws — CHANGES_REQUESTED** [review]
Errored out with
./get-library-age.sh: line 112: 1703716452 - : syntax error: operand expected (error token is "- ")

**Comment by kyallanum-MND** [comment]
Going to delete this script as it does not seem to work at the scale needed and develop in Python.

**Review by samq-ws — CHANGES_REQUESTED** [review]
Due to scale being an issue with this script.  I would add an option to run for a specific product.

File: Scripts/Mend SCA/get-library-ages.py [hunk 2]
```diff
@@ -27,73 +28,81 @@
 
 ******** Usage ********
 Make sure to install the appropriate dependencies before running this script and set all required environment variables. Afterwards, you can run the script with:
-python3 get-library-ages.py
+pip3 install requests python-dateutil
+python3 get-library-ages.py -o <output_file>.json -a <max_age_in_days>
 
-Optional Arguments: the only optional argument for this script is WS_API_KEY. If this is not specified then the Mend API will log in to the last organization accessed from the Mend Platform.
+Optional Arguments: the only optional arguments for this script are WS_API_KEY and MEND_PRODUCT_TOKEN.
```
Comment (samq-ws) [inline Scripts/Mend SCA/get-library-ages.py:34]: remove api_key and product_token and use uuids from API 2.0

File: Scripts/Mend SCA/get-library-ages.py [hunk 3]
```diff
@@ -27,73 +28,81 @@
 
 ******** Usage ********
 Make sure to install the appropriate dependencies before running this script and set all required environment variables. Afterwards, you can run the script with:
-python3 get-library-ages.py
+pip3 install requests python-dateutil
+python3 get-library-ages.py -o <output_file>.json -a <max_age_in_days>
```
Comment (samq-ws) [inline Scripts/Mend SCA/get-library-ages.py:32]: uuids should be parameters

File: Scripts/Mend SCA/get-library-ages.py [hunk 4]
```diff
@@ -27,73 +28,81 @@
...
-apt-get install python39
+apt-get install python3.9
 pip install requests python-dateutil
 export MEND_USER_KEY='<MEND_USER_KEY>'
 export MEND_URL='<MEND_URL>
 export MEND_EMAIL='<MEND_EMAIL>'
-export OUTPUT_FILE='<OUTPUT_FILE>.json'
-export MAX_LIBRARY_AGE='<MAX_AGE_IN_DAYS>'
 export WS_API_KEY='<WS_API_KEY>' <-- Optional
+export MEND_PRODUCT_TOKEN='<PRODUCT_TOKEN>' <-- Optional
```
Comment (samq-ws) [inline Scripts/Mend SCA/get-library-ages.py:45]: add a note on how to get productUUID from https://docs.mend.io/bundle/mend-api-2-0/page/index.html#tag/Entities-Organization/operation/getDomainEntities

**Review by samq-ws — CHANGES_REQUESTED** [review]

**Review by samq-ws — APPROVED** [review]

**Pull request merged**